helix-rest/src/main/java/org/apache/helix/rest/client/CustomRestClientFactory.java

```diff
@@ -33,34 +33,37 @@
 public class CustomRestClientFactory {
   private static final Logger LOG = LoggerFactory.getLogger(CustomRestClientFactory.class);
 
-  private static CustomRestClient INSTANCE = null;
+  private static HttpClient HTTP_CLIENT_INSTANCE = null;
 
   private CustomRestClientFactory() {
   }
 
+  /**
+   * Not using singleton pattern because {@link CustomRestClientImpl} has a transient memory cache with lifecycle of one
```

Comment:
Clarify this: we are not using the singleton for CustomRestClient but we are using a singleton HttpClient. The reason for not making CustomRestClient a singleton is because it contains extra member variables whose lifecycle is tied to the lifecycle of each REST request.

Reply:
The factory method itself is quite self-explanatory by reading the comments or reading the impl, don't you think so?

Reply:
The way the comment is written could be confusing because others may not know that this used to use the singleton pattern. I asked you to clarify because it wasn't self-explanatory.